addons/mail/models/mail_thread.py

```diff
@@ -2181,7 +2181,7 @@ def _notify_record_by_inbox(self, message, recipients_data, msg_vals=False, **kw
                 for channel in channels.filtered(lambda c: c.email_send):
                     users = channel.channel_partner_ids.mapped('user_ids')
                     for user in users.filtered(lambda u: u.notification_type == 'email'):
-                        channel.with_user(user).channel_seen(message.id)
+                        channel.with_user(user).sudo().channel_seen(message.id)
```

Comment:
Channel is browsed with sudo 10 lines above, the env became different here?

Also please add a test to cover this case.

Reply:
I think `with_user` drop the SU flag